src/actions/authorActions.js

```diff
@@ -0,0 +1,55 @@
+import * as types from './actionTypes';
+import authorApi from '../api/mockAuthorApi';
+import { beginAjaxCall, ajaxCallError } from './ajaxStatusActions';
+
+export function loadAuthorSuccess(authors) {
+  return { type: types.LOAD_AUTHORS_SUCCESS, authors };
+}
+
+export function createAuthorSuccess(author) {
+  return { type: types.CREATE_AUTHORS_SUCCESS, author };
+}
+
+export function updateAuthorSuccess(author) {
+  return { type: types.UPDATE_AUTHORS_SUCCESS, author };
+}
+
+export function deleteAuthorSuccess(authorId) {
+  return { type: types.DELETE_AUTHORS_SUCCESS, authorId };
+}
+
+export function loadAuthors() {
+  return function(dispatch) {
+    dispatch(beginAjaxCall());
+    return authorApi.getAllAuthors().then(authors => {
+      dispatch(loadAuthorSuccess(authors));
+    }).catch(error => {
+      throw(error);
+    });
+  };
+}
+
+export function saveAuthor(author) {
```

Comment:
Nice trick using the course id as the logic for updating or saving.
But this function name does not properly define the function the code is executing. something like
saveOrUpdateAuthor might make more sense